WordPress/openverse
# Update to Vale 3 and refactor setup in light of ov

## Fixes



Fixes https://github.com/WordPress/openverse/issues/3883 by @zackkrida

## Description



I've updated Vale to version 3. Beyond making it work (addressing breaking changes in Vale), I haven't done anything to change the Vale configuration itself. However, I did take the opportunity to remove the usage of Docker for Vale, and move it to running using the Python library, installed as a small pdm project.

This has advantages over installing it with pipx in dev_env/Dockerfile, which I write down in my updates to `.vale/README.md`. TL;DR: renovate can update it for us, we can keep a pinned version, and we can easily make sure the right version is running (without needing to tell someone to run `ov init` if some version mismatch happens).

Every time we run vale, pdm makes sure we've got the latest specified version (the one in the lock file), it syncs the styles (which is fast when they are already synced, it basically noops), and then finally actually runs Vale. I also managed to get all this working still without dumping Vale style configuration documents into the project directory by configuring Vale to use the global stylepath (but it's "global" for `ov`, so not really "global" :slightly_smiling_face:).

I had to make a small change to the heading of a file because it looks like Vale.Terms might have changed its behaviour, and it didn't like the capitalised "Decision-Making Process" for some reason :shrug: For how many things seem to have changed in Vale, that seemed like a fine thing to let go.

## Testing Instructions

Check out the branch and run `ov j .vale/run`. Confirm it runs against the same set of files as before. Checks should pass. It should also run fine in `pre-commit`, so run `ov just lint` and confirm it does!

## Checklist


- [N/A] I added or updated tests for the changes I made (if applicable).
- [N/A] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
